bootstrap.sh

```diff
@@ -0,0 +1,204 @@
+      rm "/home/root/$WGET_BINARIES_FILENAME"
+
+      cat > /home/root/.wget_bin/wget <<EOF
+#!/bin/sh
+LD_LIBRARY_PATH="/home/root/.wget_bin/dist" /home/root/.wget_bin/dist/wget \$@
+EOF
+
+      chmod +x /home/root/.wget_bin/wget
+  fi
+
+  # Ensure this binary is used
+  if [ `which wget` != '/home/root/.wget_bin/wget' ]; then
+    PATH="/home/root/.wget_bin:$PATH"
+  fi
+}
+
+
+main() {
+  # Select source for remarkable_entware repo
+  if [ -z "$REMARKABLE_ENTWARE_REPO_AUTHOR" ]; then
+    # Use LinusCDE's fork for the time beeing
+    # TODO: Switch to "Evidlo" as soon as PRs are merged
+    REMARKABLE_ENTWARE_REPO_AUTHOR=LinusCDE
+  fi
+
+
+  # Verify internet connection
+  if ! ping -c1 1.1.1.1 >/dev/null; then
+    throw "Couldn't connect to the internet. Please ensure that your reMarkable is connected to a WiFi-Network with internet access."
+  fi
+
+
+  # Verify dns
+  if ! ping -c1 remarkable.com >/dev/null; then
+    throw "Failed to resolve DNS. You seem to have internet access, but not a working DNS configuration."
+  fi
+
+
+  # Check status of installation and ensure installed
+  if [ -d /home/root/.entware ]; then
+    if [ -d /opt ] && [ `ls /opt | wc -l` -gt 0 ]; then # non-empty /opt found
+      # Entware is installed and active
+      log INFO "Entware is installed and active"
+    else
+      # Installed but not active
+      log INFO "Entware is installed but not reenabled"
+      entware_reenable
+    fi
+  else # = not found: /home/root/.entware
+    # Not installed at all
+    log INFO "Entware is not installed"
+    entware_install
+  fi
+
+  # Ensure opkg is in PATH
+  ensure_opkg_available
+  # Ensure wget is installed as opkg package
+  ensure_opt_wget_and_ca_certs_installed
+  # Add toltec repo if mising
+  ensure_toltec_repo_added
+
+  if [ $# -ge 1 ]; then
+    install_packages $@ || throw "Failed to install requested packages"
+  fi
+}
+
+
+entware_install() {
+  bootstrap_wget || throw "bootstrap_wget() failed!"
+  (wget -O- "https://raw.githubusercontent.com/$REMARKABLE_ENTWARE_REPO_AUTHOR/remarkable_entware/master/entware_install.sh" | sh) \
+    || throw "Failed to download and/or install remarkable_entware."
+  log INFO "Entware was installed successfully. PLEASE FOLLOW THE INSTRUCTIONS ABOVE."
+}
+
+
+entware_reenable() {
+  bootstrap_wget || throw "bootstrap_wget() failed!"
+  (wget -O- "https://raw.githubusercontent.com/$REMARKABLE_ENTWARE_REPO_AUTHOR/remarkable_entware/master/entware_reenable.sh" | sh) \
+    || throw "Failed to download and/or install remarkable_entware."
+  log INFO "Entware was reenabled successfully."
+}
+
+
+ensure_opkg_available() {
+  if ! which opkg >/dev/null 2>&1; then # opkg command not found
+    log WARN "Failed to find opkg command." \
+        "" \
+        "Pleae add /opt/bin & /opt/sbin to your PATH by executing" \
+        'ssh root@10.11.99.1 echo '\\\'\''PATH=/opt/bin:/opt/sbin:$PATH'\'\\\'\'' >> ~/.bashrc'\' \
+        ""
+  fi
```

Comment:
this should be automatic if /opt/bin exists (don't ask user to run the command by hand)

Reply:
Agreed. I'll add that.